packages/compiler-core/__tests__/transforms/transformElement.spec.ts

```diff
@@ -717,4 +718,42 @@ describe('compiler: element transform', () => {
       expect(vnodeCall.arguments[3]).toBe(genFlagText(PatchFlags.NEED_PATCH))
     })
   })
+
+  describe('dynamic component', () => {
+    test('static binding', () => {
+      const { node, root } = parseWithBind(`<component is="foo" />`)
+      expect(root.helpers).not.toContain(RESOLVE_DYNAMIC_COMPONENT)
+      expect(node).toMatchObject({
+        callee: CREATE_VNODE,
+        arguments: ['_component_foo']
+      })
+    })
+
+    test('dynamic binding', () => {
+      const { node, root } = parseWithBind(`<component :is="foo" />`)
+      expect(root.helpers).toContain(RESOLVE_DYNAMIC_COMPONENT)
+      expect(node.arguments).toMatchObject([
+        {
+          callee: RESOLVE_DYNAMIC_COMPONENT,
+          arguments: [
+            {
+              type: NodeTypes.SIMPLE_EXPRESSION,
+              content: 'foo'
+            }
+          ]
+        }
+      ])
+    })
+
+    test('empty "is" prop', () => {
+      const result = {
+        callee: CREATE_VNODE,
+        arguments: ['"component"']
```

Comment:
The arguments I see in template-explorer is "_component_foo"... I may have missed a transform in compiler options.

Reply:
removed the test altogether - 
`<component :is />` is now compilation error
`<component is />` still compiles, but should probably be checked along with `key`, `ref`, etc. in a central place.